### ISSUE TITLE
Codesign conflicts with use "--key" (pycrypto)

### ISSUE THREAD

**Issue by zhangxiansheng** [pyinstaller/pyinstaller#6619]
## Codesign conflicts with use "--key" (pycrypto)

* Output of `pyinstaller --version`: ```4.3 and 4.4~4.9```
* Version of Python: for example 3.6.8 or 3.7.3
* Platform: MacOS 11 (Big Sur)
* How you installed Python:  download installer from python.org
* CPU: x86_64
* I have try MacOS 10.13 and MacOS 11, MacOS 10.13 is right, but MacOS 11 will show "bootloader" bug.

App in MacOS 10.13 run with no bug.
App in MacOS 11 can not run, with bug like this:
```
Traceback (most recent call last):
  File "PyInstaller/loader/pyiboot01_bootstrap.py", line 123, in <module>
  File "PyInstaller/loader/pyimod04_ctypes.py", line 28, in install
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "ctypes/__init__.py", line 543, in <module>
  File "ctypes/__init__.py", line 273, in _reset_cache
MemoryError
[7568] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception: 
[7568] Traceback:
Traceback (most recent call last):
  File "PyInstaller/loader/pyiboot01_bootstrap.py", line 123, in <module>
  File "PyInstaller/loader/pyimod04_ctypes.py", line 28, in install
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "ctypes/__init__.py", line 543, in <module>
  File "ctypes/__init__.py", line 273, in _reset_cache
MemoryError
```

To find where is the bug, I try this with pyinstaller 4.3 and codesign the app manually.
But I found the bug is from use "--key" arg.
If I didn't use "--key xxxxx", it will work after codesigning the app.

So I have tried to use pycryptodome, pycryptodomex instead of pycrypto, all of these won't work.

I have no way to fix this bug.

**Comment by rokm** [comment]
What exactly are you doing? Are you building an application on macOS 10.13 and then trying to run that build on both 10.13 and 11, or are you building and running separately on each OS version?

Is this a onefile or a onedir build? Is it a POSIX or an .app bundle? I.e., what is the PyInstaller command you use to build the program?

**Comment by zhangxiansheng** [comment]
@rokm I tried build an application on MacOS 10.13  and run the application both on MacOS 10 and MacOS 11.  If I didn't codesign the application, they both work. But if I codesign it, on Mac 10 work,  on Mac 11 didn't work( show bootloader error).

So I tried build on MacOS 11, want the app works both on MacOS 11 and Mac 10. The same results.

To know why fail, I try this thing from using pyinstaller 4.9 to pyinstaller 4.3.

pyinstaller 4.3 is special. If use 4.3, I have to codesign the app manually.
To codesign the app manually is better to catch where is wrong.

So I got that the true problem is that codesign conflicts with use "--key" .

for example, when I use pyinstaller 4.3 to codesign the app manually.

```
pyinstaller -w -D --key XXXX  --osx-bundle-identifier XXXX program.py
```

The app can run, 
when I codesign it, it can not run.

But when not using "--key" args,  the app can run,
when codesign it, it also can run.

_"App bundle" is not about this problem,  and is about the notarization._
